assets/js/hooks/cell_editor/live_editor.js

```diff
@@ -166,6 +166,25 @@ export default class LiveEditor {
     return node.parentElement;
   }
 
+  /**
+   * Returns the current main cursor position.
+   */
+  getCurrentCursorPosition() {
+    if (!this.isMounted()) {
+      return null;
+    }
+
+    const pos = this.view.state.selection.main.head;
+    const line = this.view.state.doc.lineAt(pos);
+    const lineLength = line.to - line.from;
+    const column = pos - line.from;
+
+    if (column < 1 || column > lineLength)
+      return { line: line.number, column: 1 };
+
+    return { line: line.number, column };
```

Comment:
So I think we can just do this:

```suggestion
    const offset = pos - line.from;

    return { line: line.number, offset };
```